src/pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapPlot.jsx

```diff
@@ -274,10 +306,13 @@ const HeatmapPlot = (props) => {
     );
   }
 
+  let specCopy = { ...spec };
+  specCopy.legends = legendIsVisible ? spec.legends : [];
+
```

Comment:
will have to take a look at this on Monday.

Reply:
This is not correct use of React, and causes extraneous render steps when none is required. We should get on a call to discuss.